File: python/data_wilayah_indonesia/src/data_wilayah_indonesia/core/entities/village.py
from dataclasses import dataclass
from enum import Enum
from .administrative_unit import AdministrativeUnit

class VillageType(Enum):
    DESA = "DESA"
    KELURAHAN = "KELURAHAN"

@dataclass(frozen=True)
class Village(AdministrativeUnit):
    """
    Entitas untuk tingkat Desa/Kelurahan.
    """
    district_code: str
    type: VillageType

    def __post_init__(self):
        super().__post_init__()
        if not self.district_code:
            raise ValueError("District code tidak boleh kosong.")
        if not isinstance(self.type, VillageType):
            raise ValueError("Village type tidak valid.")

File: packages/data-wilayah-indonesia/src/core/entities/village.ts
import { AdministrativeUnit, VillageType } from "./administrative-unit";

/**
 * Village entity.
 */
export class Village extends AdministrativeUnit {
  constructor(
    code: string,
    name: string,
    public readonly districtCode: string,
    public readonly type: VillageType
  ) {
    super(code, name);
    if (!districtCode) throw new Error("District code cannot be empty.");
  }
}
